### PR TITLE
Docs revamp: polishing landing docs.

### PR DESCRIPTION
**Pull request by concretevitamin**
Preview: https://skypilot.readthedocs.io/en/docs-revamp-tmp2/docs/index.html

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

**Review by romilbhardwaj — APPROVED** [review]
Thanks @concretevitamin

File: README.md [hunk 1]
```diff
@@ -177,6 +179,8 @@ Runnable examples:
...
 
+Case Studies and Integrations: [Community Spotlights](https://blog.skypilot.co/community/)
+
```
Comment (landscapepainter) [inline README.md:183]: nit: We can perhaps sublist the `([Introductory blog post](https://blog.skypilot.co/introducing-skypilot/))` at L187 under the bullet point `[SkyPilot Blog](https://blog.skypilot.co/)` just like how it's done for the bulletpoint for `LLMs on Skypilot` and its sublist.

File: README.md [hunk 2]
```diff
@@ -177,6 +179,8 @@ Runnable examples:
...
 
+Case Studies and Integrations: [Community Spotlights](https://blog.skypilot.co/community/)
```
Comment (landscapepainter) [inline README.md:182]: nit: We can perhaps sublist the `([Introductory blog post](https://blog.skypilot.co/introducing-skypilot/))` at L187 under the bullet point `[SkyPilot Blog](https://blog.skypilot.co/)` just like how it's done for the bulletpoint for `LLMs on Skypilot` and its sublist.

File: README.md [hunk 3]
```diff
@@ -177,6 +179,8 @@ Runnable examples:
   - Add yours here & see more in [`llm/`](./llm)!
```
Comment (landscapepainter) [inline README.md:179]:
```suggestion
  - Add yours here & see more in [`llm/`](./llm)
```
nit: '/!' doesn't look clean.

File: README.md [hunk 4]
```diff
@@ -38,11 +37,11 @@
 - [Nov, 2023] Using [**Axolotl**](https://github.com/OpenAccess-AI-Collective/axolotl) to finetune Mistral 7B on the cloud (on-demand and spot): [**example**](./llm/axolotl/)
 - [Sep, 2023] Case study: [**Covariant**](https://covariant.ai/) transformed AI development on the cloud using SkyPilot, delivering models 4x faster cost-effectively: [**read the case study**](https://blog.skypilot.co/covariant/)
 - [Aug, 2023] **Finetuning Cookbook**: Finetuning Llama 2 in your own cloud environment, privately: [**example**](./llm/vicuna-llama-2/), [**blog post**](https://blog.skypilot.co/finetuning-llama2-operational-guide/)
-- [June, 2023] Serving LLM 24x Faster On the Cloud [**with vLLM**](https://vllm.ai/) and SkyPilot: [**example**](./llm/vllm/), [**blog post**](https://blog.skypilot.co/serving-llm-24x-faster-on-the-cloud-with-vllm-and-skypilot/)
 
 <details>
   <summary>Archived</summary>
 
+- [June, 2023] Serving LLM 24x Faster On the Cloud [**with vLLM**](https://vllm.ai/) and SkyPilot: [**example**](./llm/vllm/), [**blog post**](https://blog.skypilot.co/serving-llm-24x-faster-on-the-cloud-with-vllm-and-skypilot/)
```
Comment (Michaelvll) [inline README.md:44]: Do we want to keep this chronological order?

File: docs/source/docs/index.rst [hunk 5]
```diff
@@ -58,10 +59,18 @@ Current supported providers (AWS, GCP, Azure, OCI, Lambda Cloud, RunPod, Fluidst
    </picture>
    </p>
 
-More Information
---------------------------
+Ready to get started?
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply docs/source/docs/index.rst:62]: Slightly leaning towards `Getting started` as well, as it is more standard in other library's docs.

**Review by Michaelvll — APPROVED** [review]
LGTM. Thanks @concretevitamin !

File: README.md (hunk 3 above)
Comment (concretevitamin) [reply README.md:179]: Maybe fine?

File: README.md (hunk 2 above)
Comment (concretevitamin) [reply README.md:182]: Intention was to keep it that page shorter

File: docs/source/docs/index.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (concretevitamin) [reply docs/source/docs/index.rst:62]: Want to experiment with this question tone.

File: README.md (hunk 4 above)
Comment (concretevitamin) [reply README.md]: Good catch, done.

**Pull request merged**